Unnecessary dependency gsettings-desktop-schemas -> python@2

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Install formula gsettings-desktop-schemas in Travis CI environment
- What happened (include command output)
This formula is trying to install python@2 as dependency.
- What you expected to happen
Should not require python@2 dependency. There are no python scripts or modules installed by this formula, python is not even build dependency.
- Step-by-step reproduction instructions (by running `brew install` commands)
`brew install gsettings-desktop-schemas`

It looks like `python@2` is an indirect dependency, at least at build time. This may be because `gsettings-desktop-schemas` depends on `gobject-introspection`, and Homebrew's `gobject-introspection` is built against `python@2`. But we may be able to remove it from `gsettings-desktop-schemas` itself, so bottle users don't need it.

The dependency was introduced back in https://github.com/Homebrew/homebrew-core/commit/d768c47134b5d31f3e5be9579f9068673962ce43.

If I install `gobject-introspection` and then remove `python@2` and remove the `python@2` dependency from the `gsettings-desktop-schema` formula, then `brew install -s gsettings-desktop-schemas` fails with the following:

```
$ brew install -s gsettings-desktop-schemas
==> Downloading https://download.gnome.org/sources/gsettings-desktop-schemas/3.28/gsettings-desktop-schemas-3.28.
Already downloaded: /Users/janke/Library/Caches/Homebrew/gsettings-desktop-schemas-3.28.0.tar.xz
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/gsettings-desktop-schemas/3.28.0 --disable-sche
==> make install
Last 15 lines from /Users/janke/Library/Logs/Homebrew/gsettings-desktop-schemas/02.make:
2018-05-26 03:58:14 -0400

make
install

Making install in headers
CPPFLAGS="" CFLAGS="-g -O2" LDFLAGS="" CC="clang" PKG_CONFIG="/usr/local/opt/pkg-config/bin/pkg-config" GI_HOST_OS="" DLLTOOL="" CC="clang" /usr/local/Cellar/gobject-introspection/1.56.1/bin/g-ir-scanner  --add-include-path=. --namespace=GDesktopEnums --nsversion=3.0        --header-only --identifier-prefix=GDesktop --cflags-begin -I. --cflags-end  gdesktop-enums.h --output GDesktopEnums-3.0.gir
env: /usr/local/opt/python@2/bin/python2: No such file or directory
make[1]: *** [GDesktopEnums-3.0.gir] Error 127
make: *** [install-recursive] Error 1
```

If you inspect that `g-ir-scanner` command from `gobject-introspection`, you can see it's a Python program that depends on Homebrew's `python@2`.

```
$ head -1 /usr/local/Cellar/gobject-introspection/1.56.1/bin/g-ir-scanner
#!/usr/bin/env /usr/local/opt/python@2/bin/python2
```

However, pouring `gsettings-desktop-schemas` from a bottle and doing `brew test gsettings-desktop-schemas` succeeds. I'm guessing that the rest of it works too, since from Googling around it looks like `gsettings` is a C program, and grepping through its installed files doesn't reveal any `python` references.

Looks like `python@2` is really just a run-time dep of `gobject-introspection`, which is a build-time tool used by `gsettings-desktop-schema`, but not a direct dependency.

So I think the direct dependency here is in fact in error; it's actually just a dependency of `gobject-introspection`, and we can change to formula to reflect that.

--- a/Formula/gsettings-desktop-schemas.rb
+++ b/Formula/gsettings-desktop-schemas.rb
@@ -17,7 +17,6 @@ class GsettingsDesktopSchemas < Formula
   depends_on "glib"
   depends_on "gettext"
   depends_on "libffi"
-  depends_on "python@2"
 
   def install
     system "./configure", "--disable-dependency-tracking",

